include/thingset/can.h

```diff
@@ -33,12 +33,28 @@ extern "C" {
  *   tgt bus: Bus number of the target node (default for single bus systems is 0x0)
  *   src bus: Bus number of the source node (default for single bus systems is 0x0)
  *
- * Control and report messages (always single-frame):
+ * Multi-frame reports:
+ *
+ *    28      26 25 24 23 20 19     16 15  13   12  11   8 7           0
+ *   +----------+-----+-----+---------+------+-----+------+-------------+
+ *   | Priority | 0x1 | res | src bus | msg# | end | seq# | source addr |
+ *   +----------+-----+-----+---------+------+-----+------+-------------+
+ *
+ *   Priority: 5 or 7
+ *   msg#: Wrapping message counter from 0 to 7
+ *   end: End of message flag
+ *   seq#: Wrapping sequence counter from 0 to 15
+ *
+ * Single-frame reports:
  *
  *    28      26 25 24 23           16 15            8 7             0
- *   +----------+-----+---------------+---------------+---------------+
- *   | Priority | 0x2 | data ID (MSB) | data ID (LSB) |  source addr  |
- *   +----------+-----+---------------+---------------+---------------+
+ *   +----------+------+---------------+---------------+---------------+
+ *   | Priority | Type | data ID (MSB) | data ID (LSB) |  source addr  |
+ *   +----------+------+---------------+---------------+---------------+
+ *
+ *   Type:
+ *     0x1: multi-frame
+ *     0x2: single-frame
```

Comment:
The challenge you have with this notation is that it could be construed to mean that if a _report_ (in the ThingSet sense) fits in one frame, it would use `0x2`; if it spans multiple frames, it would become `0x1` (as it does in the current implementation).

In fact, `0x1` means _one or more frames containing one or more items_ (aka a _report_); `0x2` means _exactly one frame containing exactly one item_ (aka an _item_).

Reply:
Oops, this was completely wrong. Single-frame reports are always of type `0x2`. And yes, `0x2` means exactly one frame. In theory it may contain more than one item (if the data object ID is a group), but that would only make sense for CAN FD with a bit more space than 8 bytes.